oml/functional/metrics.py

```diff
@@ -294,42 +294,45 @@ def calc_precision(gt_tops: Tensor, n_gt: Tensor, top_k: Tuple[int, ...]) -> Lis
             https://en.wikipedia.org/wiki/Evaluation_measures_(information_retrieval)#Precision_at_k
 
     Example:
-        >>> gt_tops = torch.tensor([
-        ...                         [1, 0],
-        ...                         [0, 1],
-        ...                         [0, 0]
-        ... ], dtype=torch.bool)
-        >>> n_gt = torch.tensor([2, 3, 5])
+
+        >>> gt_tops = [
+        ...     BoolTensor([1, 0]),
+        ...     BoolTensor([0, 1, 1]),
+        ...     BoolTensor([0, 0])
+        ... ]
+        >>> n_gt = [2, 3, 5]
         >>> calc_precision(gt_tops, n_gt, top_k=(1, 2))
         [tensor([1., 0., 0.]), tensor([0.5000, 0.5000, 0.0000])]
+
     """
     check_if_nonempty_positive_integers(top_k, "top_k")
-    top_k = _clip_max_with_warning(top_k, gt_tops.shape[1])
+
+    def precision_single(is_correct: BoolTensor, n_gt_: int, k_: int) -> float:
+        k_ = min(k_, len(is_correct))
+        value = torch.cumsum(is_correct, dim=0)[k_ - 1] / min(n_gt_, k_)
+        return float(value)
```

Comment:
I return float from the inner function because I expect here just a single value without any extra dimensions which I may have if I keep it as tensor